### DISCUSSION TITLE
How prisma handle big queries with include option

### DISCUSSION THREAD

**Discussion by primemb** [prisma/prisma#20364] [Q&A] [answered]
Hello, I'd like to confirm something regarding the scenario I'm currently working on.

in my application, i have some channels, and each channel can have many videos, so it is a one-to-many relationship. right now i want to get all videos with it channel info i do it like this : 
```
movies = await prisma.movie.findMany({
      where: { user_id: user.id },
      include: { channel:true },
    });
```

so my question is on big queries how does Prisma handle this query? let's say we have 100000 videos and only 10 channels, is Prisma cache channels and only query channel table 10 times or 100000 times? i use prisma with MongoDB by the way

**Comment by ludralph** [comment, marked as answer]
Hi @primemb 

Prisma does not cache query results. MongoDB, being a NoSQL database, typically manages its own caching and query optimization. This means that if you perform the same query multiple times, MongoDB may cache the results on its end for better performance.  If you run the given query to fetch videos along with their channel information, Prisma will make a query to the MongoDB server for each unique channel ID. If there are 10 unique channels, Prisma will send 10 queries to MongoDB, not 100,000 queries.


If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by janpio** [reply]
You can figure out how many queries Prisma executes by enabling query logging: https://pris.ly/d/logging That will then log each individual query to your console, and you can also understand what the queries (or "commands" in case of MongoDB) look like.
